src/prompt/engine.go

```diff
@@ -500,3 +509,80 @@ func (e *Engine) adjustTrailingDiamondColorOverrides() {
 		adjustOverride(match[terminal.ANCHOR], color.Ansi(match[terminal.FG]))
 	}
 }
+
+func (e *Engine) fetchPromptCache() bool {
+	data, ok := e.Env.Cache().Get(cache.PROMPTCACHE)
+	if !ok {
+		return false
+	}
+
+	e.promptCache = &promptCache{}
+	err := json.Unmarshal([]byte(data), e.promptCache)
+	if err != nil {
+		return false
+	}
+
+	e.write(e.promptCache.Prompt)
+	e.currentLineLength = e.promptCache.CurrentLineLength
+	e.rprompt = e.promptCache.RPrompt
+	e.rpromptLength = e.promptCache.RPromptLength
+
+	return true
+}
+
+func (e *Engine) updatePromptCache(value *promptCache) {
+	cacheJSON, err := json.Marshal(value)
+	if err != nil {
+		return
+	}
+	e.Env.Cache().Set(cache.PROMPTCACHE, string(cacheJSON), 1440)
+}
+
+// New returns a prompt engine initialized with the
```

Comment:
why did you move this one?

Reply:
Just an advice on the code organization. The function returns a new prompt engine and it's all about the "engine" logic, so there is no need to put it in a standalone source file, IMO. It's okay to revert this if you prefer the original way. :)

Reply:
@lewis-yeung I think we can argue either way. This is fine, I'll allow it 😃